src/fetch/rewards.py

```diff
@@ -0,0 +1,167 @@
+"""Functionality for computing solver rewards
+
+The only function used in other parts of the code is `compute_solver_rewards` for generating
+aggregate reward information per solver."""
+
+from pandas import DataFrame, Series
+from src.fetch.prices import TokenConversion
+
+EPSILON_UPPER = 12_000_000_000_000_000
+EPSILON_LOWER = 10_000_000_000_000_000
+PERIOD_BUDGET_COW = 250000 * 10**18
+CONSISTENCY_REWARD_CAP_ETH = 6 * 10**18
+QUOTE_REWARD_COW = 6 * 10**18
+QUOTE_REWARD_CAP_ETH = 6 * 10**14
+
+
+def compute_solver_rewards(
+    batch_data_df: DataFrame, trade_data_df: DataFrame, converter: TokenConversion
+) -> DataFrame:
+    """Compute solver rewards
+    The input dataframes `batch_data_df` and `trade_data_df` containd data on batches and
+    trades, respectively. It requires TokenConversion to convert betwenn ETH and COW.
+    The function creates a new dataframe with information on primary, secondary, and quote rewards.
```

Comment:
Since we usually say "performance" and "consistency/participation" rewards, maybe add a clarification here about what primary and secondary corresponds to.

Reply:
I added some clarification.